File: billing_ui_app/src/app/core/models/billing/write-off-claim-model.ts
import { UserInfo } from "./get-claim-by-identifier";

export class WriteOffClaimModelWithUserInfo extends UserInfo {
    claimId: number;
    serviceLineId: number | null;
    amountTypeId: number;
    amount: number;
    applicationTypeId: number | null;
    reasonCodeId: number;
    note: string;
    isServiceLine: boolean | null;
}

export class ClaimOrChargeToWriteOff {
    balanceAmount: number;
    claimId: number;
    chargeId: number;
}

export class AddWriteOffResponseModel {
    success : boolean;
    errorMsg : string;
}

File: Microservices/BillingService/BillingService.Domain/Models/Claims/WriteOff/WriteOffClaimModel.cs
namespace BillingService.Domain.Models.Claims
{
    public class WriteOffClaimModelWithUserInfo : UserInfo
    {
        public int ClaimId { get; set; }
        public int? ServiceLineId { get; set; }
        public int AmountTypeId { get; set; }
        public decimal Amount { get; set; }
        public int? ApplicationTypeId { get; set; }
        public int ReasonCodeId { get; set; }
        public string Note { get; set; }
        public bool? IsServiceLine { get; set; }
    }
}
